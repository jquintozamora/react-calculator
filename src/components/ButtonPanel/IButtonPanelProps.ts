export interface IButtonPanelProps {
  onButtonClick: (name?: string) => void;
  operator: string | null;
}


