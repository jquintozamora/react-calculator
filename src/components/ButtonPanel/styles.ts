import styled from "styled-components";

export const ButtonPanelContainer = styled.div`
  padding: 10px;
`;

export const ButtonPanelLine = styled.div`
  display: flex;
`;