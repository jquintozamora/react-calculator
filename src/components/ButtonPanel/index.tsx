import * as React from "react";
import Button from "../Button";
import { IButtonMouseEvent } from "../Button/IButtonProps";
import { IButtonPanelProps } from "./IButtonPanelProps";
import { ButtonPanelContainer, ButtonPanelLine } from "./styles";

import {
  black,
  eeBlue,
  eeDarkGrey,
  eeDarkYellow,
  eeLightBlue,
  eeYellow,
  lightBlack,
  white
} from "../../styles/colors";

class ButtonPanel extends React.PureComponent<IButtonPanelProps, {}> {
  public render() {
    return (
      <ButtonPanelContainer>
        <ButtonPanelLine>
          <Button
            data-name="AC"
            onClick={this.handleClick}
            bgColor={eeLightBlue}
            borderColor={eeBlue}
            color={eeDarkGrey}
            hoverColor={white}
          />
          <Button
            data-name="+/-"
            onClick={this.handleClick}
            bgColor={eeLightBlue}
            borderColor={eeBlue}
            color={eeDarkGrey}
            hoverColor={white}
          />
          <Button
            data-name="%"
            onClick={this.handleClick}
            bgColor={eeLightBlue}
            borderColor={eeBlue}
            color={eeDarkGrey}
            hoverColor={white}
          />
          <Button
            data-name="รท"
            onClick={this.handleClick}
            bgColor={eeYellow}
            color={black}
            borderColor={eeDarkYellow}
            hoverColor={black}
            selected={this.props.operator === "รท"}
          />
        </ButtonPanelLine>
        <ButtonPanelLine>
          <Button
            data-name="7"
            onClick={this.handleClick}
            bgColor={lightBlack}
            color={white}
            hoverColor={black}
          />
          <Button
            data-name="8"
            onClick={this.handleClick}
            bgColor={lightBlack}
            color={white}
            hoverColor={black}
          />
          <Button
            data-name="9"
            onClick={this.handleClick}
            bgColor={lightBlack}
            color={white}
            hoverColor={black}
          />
          <Button
            data-name="x"
            onClick={this.handleClick}
            bgColor={eeYellow}
            color={black}
            borderColor={eeDarkYellow}
            hoverColor={black}
            selected={this.props.operator === "x"}
          />
        </ButtonPanelLine>
        <ButtonPanelLine>
          <Button
            data-name="4"
            onClick={this.handleClick}
            bgColor={lightBlack}
            color={white}
            hoverColor={black}
          />
          <Button
            data-name="5"
            onClick={this.handleClick}
            bgColor={lightBlack}
            color={white}
            hoverColor={black}
          />
          <Button
            data-name="6"
            onClick={this.handleClick}
            bgColor={lightBlack}
            color={white}
            hoverColor={black}
          />
          <Button
            data-name="-"
            onClick={this.handleClick}
            bgColor={eeYellow}
            color={black}
            borderColor={eeDarkYellow}
            hoverColor={black}
            selected={this.props.operator === "-"}
          />
        </ButtonPanelLine>
        <ButtonPanelLine>
          <Button
            data-name="1"
            onClick={this.handleClick}
            bgColor={lightBlack}
            color={white}
            hoverColor={black}
          />
          <Button
            data-name="2"
            onClick={this.handleClick}
            bgColor={lightBlack}
            color={white}
            hoverColor={black}
          />
          <Button
            data-name="3"
            onClick={this.handleClick}
            bgColor={lightBlack}
            color={white}
            hoverColor={black}
          />
          <Button
            data-name="+"
            onClick={this.handleClick}
            bgColor={eeYellow}
            color={black}
            borderColor={eeDarkYellow}
            hoverColor={black}
            selected={this.props.operator === "+"}
          />
        </ButtonPanelLine>
        <ButtonPanelLine>
          <Button
            data-name="0"
            onClick={this.handleClick}
            expand={true}
            bgColor={lightBlack}
            color={white}
            hoverColor={black}
          />
          <Button data-name="." onClick={this.handleClick} bgColor={lightBlack} color={white} />
          <Button
            data-name="="
            onClick={this.handleClick}
            bgColor={eeYellow}
            color={black}
            borderColor={eeDarkYellow}
            hoverColor={black}
          />
        </ButtonPanelLine>
      </ButtonPanelContainer>
    );
  }

  private handleClick = (ev: IButtonMouseEvent) => {
    const buttonName = ev.currentTarget.dataset.name;
    if (buttonName) {
      this.props.onButtonClick(buttonName);
    }
  };
}

export default ButtonPanel;
