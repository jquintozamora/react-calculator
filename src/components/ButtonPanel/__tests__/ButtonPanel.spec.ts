import ButtonPanel from "..";
import {
  buildTestShallowRender,
  buildTestRender
} from "../../../../test/unit/helpers/ComponentHelper";

describe("<ButtonPanel />", () => {
  const renderComponent = buildTestShallowRender(ButtonPanel);
  const defaultProps = {
    onButtonClick: () => ({}),
    operator: null
  };

  describe("@renders", () => {
    it("should render default state", () => {
      expect(renderComponent().tree).toMatchSnapshot();
    });
    it("should render opacity when operator is selected", () => {
      expect(renderComponent({ ...defaultProps, operator: "+" }).tree).toMatchSnapshot();
    });
  });

  describe("@events", () => {
    const renderComponentEv = buildTestRender(ButtonPanel);
    it("should call onClick function", () => {
      const mockOnclick = jest.fn();
      const { root } = renderComponentEv({ ...defaultProps, onButtonClick: mockOnclick });
      expect(mockOnclick).toHaveBeenCalledTimes(0);
      const button = root.find(el => el.props["data-name"] === "1");
      button.props.onClick({ currentTarget: { dataset: { name: "1" } } });
      expect(mockOnclick).toHaveBeenCalledTimes(1);
      expect(mockOnclick).toHaveBeenCalledWith("1");
    });
  });
});
