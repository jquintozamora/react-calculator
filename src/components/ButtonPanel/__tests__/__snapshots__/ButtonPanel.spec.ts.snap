// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonPanel /> @renders should render default state 1`] = `
<ForwardRef>
  <ForwardRef>
    <Unknown
      bgColor="#cdeeff"
      borderColor="#1795d4"
      color="#666"
      data-name="AC"
      hoverColor="#FFFFFF"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#cdeeff"
      borderColor="#1795d4"
      color="#666"
      data-name="+/-"
      hoverColor="#FFFFFF"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#cdeeff"
      borderColor="#1795d4"
      color="#666"
      data-name="%"
      hoverColor="#FFFFFF"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#fed800"
      borderColor="#d4b526"
      color="#000000"
      data-name="รท"
      hoverColor="#000000"
      onClick={[Function]}
      selected={false}
    />
  </ForwardRef>
  <ForwardRef>
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="7"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="8"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="9"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#fed800"
      borderColor="#d4b526"
      color="#000000"
      data-name="x"
      hoverColor="#000000"
      onClick={[Function]}
      selected={false}
    />
  </ForwardRef>
  <ForwardRef>
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="4"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="5"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="6"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#fed800"
      borderColor="#d4b526"
      color="#000000"
      data-name="-"
      hoverColor="#000000"
      onClick={[Function]}
      selected={false}
    />
  </ForwardRef>
  <ForwardRef>
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="1"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="2"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="3"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#fed800"
      borderColor="#d4b526"
      color="#000000"
      data-name="+"
      hoverColor="#000000"
      onClick={[Function]}
      selected={false}
    />
  </ForwardRef>
  <ForwardRef>
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="0"
      expand={true}
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="."
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#fed800"
      borderColor="#d4b526"
      color="#000000"
      data-name="="
      hoverColor="#000000"
      onClick={[Function]}
    />
  </ForwardRef>
</ForwardRef>
`;

exports[`<ButtonPanel /> @renders should render opacity when operator is selected 1`] = `
<ForwardRef>
  <ForwardRef>
    <Unknown
      bgColor="#cdeeff"
      borderColor="#1795d4"
      color="#666"
      data-name="AC"
      hoverColor="#FFFFFF"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#cdeeff"
      borderColor="#1795d4"
      color="#666"
      data-name="+/-"
      hoverColor="#FFFFFF"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#cdeeff"
      borderColor="#1795d4"
      color="#666"
      data-name="%"
      hoverColor="#FFFFFF"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#fed800"
      borderColor="#d4b526"
      color="#000000"
      data-name="รท"
      hoverColor="#000000"
      onClick={[Function]}
      selected={false}
    />
  </ForwardRef>
  <ForwardRef>
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="7"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="8"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="9"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#fed800"
      borderColor="#d4b526"
      color="#000000"
      data-name="x"
      hoverColor="#000000"
      onClick={[Function]}
      selected={false}
    />
  </ForwardRef>
  <ForwardRef>
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="4"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="5"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="6"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#fed800"
      borderColor="#d4b526"
      color="#000000"
      data-name="-"
      hoverColor="#000000"
      onClick={[Function]}
      selected={false}
    />
  </ForwardRef>
  <ForwardRef>
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="1"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="2"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="3"
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#fed800"
      borderColor="#d4b526"
      color="#000000"
      data-name="+"
      hoverColor="#000000"
      onClick={[Function]}
      selected={true}
    />
  </ForwardRef>
  <ForwardRef>
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="0"
      expand={true}
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#373d45"
      borderColor="#949494"
      color="#FFFFFF"
      data-name="."
      hoverColor="#000000"
      onClick={[Function]}
    />
    <Unknown
      bgColor="#fed800"
      borderColor="#d4b526"
      color="#000000"
      data-name="="
      hoverColor="#000000"
      onClick={[Function]}
    />
  </ForwardRef>
</ForwardRef>
`;
